src/main/java/com/jpr/maintenance/database/service/UserService.java

```diff
@@ -2,23 +2,64 @@
 
 import com.jpr.maintenance.database.model.UserEntity;
 import com.jpr.maintenance.database.repository.UserRepository;
+import com.jpr.maintenance.graphql.model.FindUserInput;
 import com.jpr.maintenance.graphql.model.UserOutput;
-import com.jpr.maintenance.validation.model.User;
+import com.jpr.maintenance.validation.ValidationService;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
 
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
 @Service
 @RequiredArgsConstructor
 public class UserService {
     private final UserRepository userRepository;
 
-    public Mono<UserOutput> findByUser(final User user) {
-        return userRepository.findByUserName(user.username())
-            .flatMap(u -> u.verify(user))
+    public Mono<UserOutput> findByUserName(final FindUserInput input) {
+        return ValidationService.validate(input)
+            .flatMap(i -> userRepository.findByUserName(i.username()))
             .map(UserOutput::of);
     }
 
+
+
+    public static void main(String... args) {
```

Comment:
testing purposes?

Reply:
Haha, whoops, forgot to delete this before pushing. Indeed, was for playing around with Collectors.groupingBy.